### PR TITLE
update as to 10, add wasi, add sandbox

### PR DESCRIPTION
**Pull request by jtenner**
Finally, WASI support.

### CODE REVIEW

**Review by MaxGraey — APPROVED** [review]

File: tests/node.js [hunk 1]
```diff
@@ -122,24 +121,27 @@ function runTest(fileName, type, binary, wat) {
   // should not block testing
   promises.push(fs.writeFile(watPath, wat));
 
-  const context = new TestContext({
-    fileName: fileNamePath, // set the fileName
-    reporter, // use verbose reporter
-    binary, // pass the binary to get function names
-  });
   const wasi = new WASI({
     args: [],
```
Comment (MaxGraey) [inline tests/node.js]:
```suggestion
    args: process.argv,
```

Comment (jtenner) [reply tests/node.js]: I believe we should actually make this configurable via `testname.spec.wasi.js` which would be shaped something like this:

```js
module.exports = {
  env: {},
  argv: [],
  preopens: {}
}
```

Do you think this is a better idea?

Comment (MaxGraey) [reply tests/node.js]: env / argv just pass from node.js environment. `preopens` could be configured via cli argument like `--preopens="file1, file2 ..."`. Also you could add some `json` config as option

Comment (jtenner) [reply tests/node.js]: Why not both? And a `json` configuration would probably not work well, because we will want access to `process.env` likely.

Comment (MaxGraey) [reply tests/node.js]: It's up to you. But if you want support many ways you should decide overwriting order or union instead. So it becomes more complicated

Comment (jtenner) [reply tests/node.js]: I went with `--preopens dir1=dir1,dir2=dir2` etc.

And `{testname}.spec.wasi.js` will override the default configuration completely.

**Review by dcodeIO — APPROVED** [review]

**Pull request merged**